src/manifest.rs

```diff
@@ -286,21 +286,34 @@ impl LocalManifest {
         })
     }
 
-    /// Lookup a dependency
-    pub fn get_dependency_versions<'s>(
+    fn filter_dependencies<'s, P>(
```

Comment:
Nit: I recommend putting implementation details after their relevant public functions.  When someone is jumping into a file, they generally start top-bottom and having pub items earlier gives them the context to decide how much they are to dig into details.